modules/aggregation-layers/src/aggregation-layer.js

```diff
@@ -18,26 +18,14 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-import {CompositeLayer, AttributeManager, experimental} from '@deck.gl/core';
+import {CompositeLayer, AttributeManager, experimental, Layer} from '@deck.gl/core';
 import {cssToDeviceRatio, log} from '@luma.gl/core';
 const {compareProps} = experimental;
 
-// props when changed results in new uniforms that requires re-aggregation
-const UNIFORM_PROPS = [
-  // DATA-FILTER extension
-  'filterEnabled',
-  'filterRange',
-  'filterSoftRange',
-  'filterTransformSize',
-  'filterTransformColor'
-];
-
 export default class AggregationLayer extends CompositeLayer {
-  initializeState(aggregationProps = []) {
+  initializeState(ignoreProps = {}) {
```

Comment:
You are filtering `defaultProps` in each sub layer to generate this argument. There are several potential issues:
- `defaultProps` may contain `[key]: 0` or `[key]: null` which will defeat the check in `compareProps`
- we may have multiple layers of inheritance in the future, and this process only accounts for the leaf class and the base Layer

If you instead pass the aggregation props array here, you can perform the filtering with `filterProps(this.constructor._propTypes, aggregationProps)`. `constructor._propTypes` is always normalized and merged up the inheritance chain.